packages/app/webpack.config.js

```diff
@@ -24,6 +26,18 @@ const getSegmentKey = () => {
   return 'Cs2gImUHsghl4SZD8GB1xyFs23oaNAGa';
 };
 
+const getVersion = async () => {
+  const exec = promisify(_exec);
+  const branch = await exec(`git rev-parse --abbrev-ref HEAD | cut -d'/' -f2`);
+  if (branch.stderr) {
+    return version;
+  }
+  if (!branch.stdout.includes('master')) {
+    return `${version}.${Math.floor(Math.floor(Math.random() * 1000))}`;
```

Comment:
To make the version more identifiable, should we instead be appending the branch name to the version rather than a random number? Something like `return `${version}.${branch}`. Not sure if this information can easily be obtained here, if it this is not that important from a dev perspective. Just thinking it may be difficult to associate a particular non-master build to a branch or PR.

Reply:
I rebased my older commits - but that's exactly what I had! The version string would be `${version}-${branch.split('/').last}-${shortSha}`. However, Firefox is really strict about version numbers. There can only be four parts, and they each have to have a leading number (non-zero), and no dashes are allowed.

I've had to read through a bunch of their validation code, you can see [examples on their Github](https://github.com/mozilla/addons-linter/blob/master/tests/unit/schema/test.formats.js#L12). Kind of annoying :)

Reply:
That's so strange and arbitrary. Why is `'1.01a'` valid but `'1.01'` is not? I see your reasoning, thanks for the clarification.